File: 2025/SoyoungJung/week10/2025-08-10-Climbing-Stairs.ts
function climbStairs(n: number): number {
  if (n <= 2) return n;

  let prev2 = 1,
    prev1 = 2;
  for (let i = 3; i <= n; i++) {
    const temp = prev1;
    prev1 = prev1 + prev2;
    prev2 = temp;
  }
  return prev1;
}

File: 2025/HyeGyung/2025-08-10-Climbing-Stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= 1:
            return 1
        prev2, prev1 = 1, 1  
        for _ in range(2, n + 1):
            prev2, prev1 = prev1, prev1 + prev2
        return prev1
